### DISCUSSION TITLE
How does Remote Logging establish the connection between app and device?

### DISCUSSION THREAD

**Discussion by johndkouris** [kean/Pulse#142] [Q&A] [answered]
Hi!

I'm looking into implementing Pulse within my company's app. A question from security that came up is how the remote logging works under-the-hood? 

Is anyone able to explain how the connection is made and how secure it is? 

Thanks in advance!

**Comment by kean** [comment, marked as answer]
It uses Bonjour to establish a connection on a local network and uses TCP sockets to transfer messages (binary).

**Reply by johndkouris** [reply]
Thank you @kean!
